zero_bin/common/src/prover_state/persistence.rs

```diff
@@ -277,11 +289,18 @@ fn prover_to_disk(
     Ok(())
 }
 
-/// If we're running in the cargo workspace, then always use the `circuits`
-/// directory that lives in `tools/`. Otherwise, just use `circuits` in the
-/// current directory.
-fn relative_circuit_dir_path() -> String {
-    env::var(CARGO_WORKSPACE_DIR_ENV)
-        .map(|p| format!("{}/{}", p, CIRCUITS_DIR))
-        .unwrap_or_else(|_| CIRCUITS_DIR.to_string())
+/// We store serialized circuits inside the cache directory specified by an env
+/// variable. If the user does not set this, then we set it base to the OS's
+/// standard location for the cache directory.
+pub(crate) fn set_circuit_cache_dir_env_if_not_set() -> anyhow::Result<()> {
```

Comment:
If we are good to have this in `common`, then I think this is probably the best place for this function. Other choice is `lib.rs`.

Reply:
Regarding naming, this is already implied by the hardcoded constants + doc that we intend to specify the dir env at `ZK_EVM_CACHE_DIR_ENV`, so I would simplify the name of the method to `set_circuit_cache_dir_env`

Reply:
Oh so just to be clear, the user _could_ specify the `env` variable on their own and this logic would check if it's set and not set it, so there are cases where the `env` variable is already set.

Reply:
If the user is to specify a different location, what's the rationale with not updating the cache location? Fallbacks to the user to not do it multiple times

Reply:
I might be misunderstanding something, but if the user has already set the env var, there's nothing to do since the circuit directory logic later reads directly from the env var as well.